software_checklist.md

```diff
@@ -0,0 +1,141 @@
+
+### Are there appropriate tests?  (And are they automated?)
+
+- Are there unit tests? What is covered?
+- System and integration tests?  Automated model validation tests?
+- Regression tests? (Which show whether changes to the code lead to changes in the output. Changes to the model will be expected to change the output, but many other changes, such as refactoring and adding new features, should not. Having these tests gives confidence that the code hasn't developed bugs due to unintentional changes.)
+- Is there CI?
+- Is everything you need to run the tests (including documentation) in the repository (or the data pipeline where appropriate)?
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> [Unit tests and regression tests](README.md#Tests) are present ([![Code Coverage](https://codecov.io/github/ScottishCovidResponse/simple_network_sim/coverage.svg?branch=master&token=)](https://codecov.io/gh/ScottishCovidResponse/simple_network_sim)), run via continuous integration [![Build Status](https://travis-ci.org/ScottishCovidResponse/simple_network_sim.svg?branch=master)](https://travis-ci.org/ScottishCovidResponse/simple_network_sim). Automated model validation tests are not, but could be added. Jupyter notebooks in the repository are not currently tested - this is difficult at present as it relies on data we cannot share and is computationally expensive - ultimately the notebooks should at least be run end-to-end as part of testing.
+
+### Are the scientific results of runs robust to different ways of running the code?
+
+- Running on a different machine?
+- With different number of processes?
+- With different compilers and optimisation levels?
+- Running in debug mode?
+
+(We don't require bitwise identical results here, but the broad conclusions after looking at the results of the test case should be the same.) 
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> This applies to our test data and (can’t guarantee that it will run on any system) but regression tests have been run (informally) on a range of platforms (e.g. HPC, Windows 10, OSX). We use multiple processes only to run the stochastic model multiple times with different random seeds, not to accelerate an single model run.
```

Comment:
@aflag and @FiodorG - have I described this correctly, please?

Reply:
this is indeed correct